apps/ui/src/context/config.tsx

```diff
@@ -17,20 +20,36 @@ export function ConfigProvider({
   const [backendURL, setBackendURL] = useState(
     localStorage.getItem("backendURL") || "http://localhost:16686",
   );
-
-  const handleSetBackendURL = useCallback(
-    (backendURL: string) => {
-      localStorage.setItem("backendURL", backendURL);
-      setBackendURL(backendURL);
-    },
-    [setBackendURL],
+  const [routes, setRoutes] = useState<string[]>(
+    JSON.parse(localStorage.getItem("routes") || "[]"),
   );
 
+  const handleEnableRoute = (type: string) => {
+    localStorage.setItem("routes", JSON.stringify([...routes, type]));
+    setRoutes([...routes, type]);
```

Comment:
```suggestion
    setRoutes((oldRoutes) => [...oldRoutes, types]);
```

Should we be using the setState callback here to ensure we merge the asynchronous state in a safe manner?